### ISSUE TITLE
GLFW error 65543

### ISSUE THREAD

**Issue by HalJordan** [PrismLauncher/PrismLauncher#954]
### Version of Prism Launcher

6.3

### Version of Qt

6.4.3

### Description of bug

Suddenly getting:

```
GLFW error 65543: GLX: Failed to create context: GLXBadFBConfig.

Please make sure you have up-to-date drivers (see aka.ms/mcdriver for instructions).
```
I've tried multiple packs from multiple sources(ftb, curse, etc), rebooted. The whole nine yards. I've reinstalled any graphics drivers/libraries, mesa, et al. It only happens with Prism. Atlauncher and other games work without issue which is what's puzzling and why I'm coming here.

I've also tried with the built in and native GLFW.

### Steps to reproduce

Try to start any instance.

### This issue is unique

- [X] I have searched the issue tracker and did not find an issue describing my bug.

**Comment by ghost** [comment]
Same for me
```
[15:10:06] [Render thread/WARN]: Failed to create window: 
net.minecraft.class_1041$class_4716: **GLFW error 65545: GLX: Failed to find a suitable GLXFBConfig**
	at net.minecraft.class_1041.method_4501(class_1041.java:223) ~[client-intermediary.jar:?]
	at org.lwjgl.glfw.GLFWErrorCallbackI.callback(GLFWErrorCallbackI.java:43) ~[lwjgl-glfw-3.3.1.jar:?]
	at org.lwjgl.system.JNI.invokePPPP(Native Method) ~[lwjgl-3.3.1.jar:?]
	at org.lwjgl.glfw.GLFW.nglfwCreateWindow(GLFW.java:2024) ~[lwjgl-glfw-3.3.1.jar:?]
	at org.lwjgl.glfw.GLFW.glfwCreateWindow(GLFW.java:2197) ~[lwjgl-glfw-3.3.1.jar:?]
	at net.minecraft.class_1041.<init>(class_1041.java:93) ~[client-intermediary.jar:?]
	at net.minecraft.class_3682.method_16038(class_3682.java:21) ~[client-intermediary.jar:?]
	at net.minecraft.class_310.<init>(class_310.java:484) ~[client-intermediary.jar:?]
	at net.minecraft.client.main.Main.method_44604(Main.java:205) [client-intermediary.jar:?]
	at net.minecraft.client.main.Main.main(Main.java:51) [client-intermediary.jar:?]
	at net.fabricmc.loader.impl.game.minecraft.MinecraftGameProvider.launch(MinecraftGameProvider.java:462) [fabric-loader-0.14.17.jar:?]
	at net.fabricmc.loader.impl.launch.knot.Knot.launch(Knot.java:74) [fabric-loader-0.14.17.jar:?]
	at net.fabricmc.loader.impl.launch.knot.KnotClient.main(KnotClient.java:23) [fabric-loader-0.14.17.jar:?]
	at org.prismlauncher.launcher.impl.StandardLauncher.launch(StandardLauncher.java:88) [NewLaunch.jar:?]
	at org.prismlauncher.EntryPoint.listen(EntryPoint.java:126) [NewLaunch.jar:?]
	at org.prismlauncher.EntryPoint.main(EntryPoint.java:71) [NewLaunch.jar:?]
Process was killed by user.

```

**Comment by ghost** [comment]
![image](https://user-images.githubusercontent.com/76007630/230920370-ed0755d4-1397-4e29-80a9-53cc34b70cec.png)

**Comment by ghost** [comment]
However doing a complete reinstall and deleting the configs made in `~/.var` made it work again. (Flatpak)

**Comment by Vinesauced** [comment]
Having the same issue, but with one specific modpack on modrinth, [Pack Zero](https://modrinth.com/mod/xmdQgHni).
I'm using a patched version of glfw, called glfw-wayland-minecraft-libdecoration in the Arch repos on Garuda Linux. I can post the full crash report if it helps.

**Comment by tabbeddev** [comment]
If you get an error like this, try to restart your computer first. It could have been caused by a driver update.

```
GLFW error 65543: GLX: Failed to create context: BadValue (integer parameter out of range for operation).
Please make sure you have up-to-date drivers (see aka.ms/mcdriver for instructions).
```

Edit: replaced my old terrible english skills with less terrible english skills.

**Comment by akkaraul** [comment]
Good luck, my hardware doesn't support OpenGL and I won't be able to play the new modpack: DawnCraft, I'm desperate

**Comment by orowith2os** [comment]
For those using a patched GLFW, see https://github.com/CaffeineMC/sodium-fabric/issues/1624/

**Comment by 46620** [comment]
~~Same issue. Using an Intel Arc A770 on Arch Linux. Only had this issue after a BIOS updating and turning ResizeBAR on.~~

~~[Log file](https://mclo.gs/JeVnyVm)~~

Either a driver update in the latest kernel update or disabiling Distant Horizons fixed it for me.

**Comment by ghost** [comment]
This issue is not specific to nvidia. Updating to prismlauncer 8.x is when this issue introduced itself on my end. Deleting the current configuration and reinstalling prism indeed "fixes" the issue however this means losing all current data. I'm currently already on unstable for gpu drivers and can not change to a new version unless I'm expected to run a testing branch (not going to happen). This is definitely a configuration issue during upgrade from 7.x -> 8.x of prism.

To be clear, I'm using X11 and not wayland. I am also not using a custom modified glfw but what ships directly from the gentoo repos.

glxinfo:
GLX version: 1.4
    Vendor: AMD (0x1002)
    Device: AMD Radeon RX 6800 XT (radeonsi, navi21, LLVM 16.0.6, DRM 3.54, 6.6.1-gentoo-dist) (0x73bf)
    Version: 23.3.0
    Max core profile version: 4.6
    Max compat profile version: 4.6
    Max GLES1 profile version: 1.1
    Max GLES[23] profile version: 3.2

lspci:

0c:00.0 VGA compatible controller: Advanced Micro Devices, Inc. [AMD/ATI] Navi 21 [Radeon RX 6800/6800 XT / 6900 XT] (rev c1)
        Subsystem: Tul Corporation / PowerColor Navi 21 [Radeon RX 6800/6800 XT / 6900 XT]
        Kernel driver in use: amdgpu

**Comment by Scrumplex** [comment]
A simple solution might be not to use an unstable glfw.
LWJGL does not use glfw's stable branch so running Minecraft with a stable GLFW will cause issues.

The reason why this problem appeared with the update to 8.0, is that the setting didn't work for Minecraft versions 1.19+ (or more specifically LWJGL 3.3.1+) prior to Prism Launcher 8.0

See https://github.com/PrismLauncher/PrismLauncher/pull/1392 and https://github.com/PrismLauncher/PrismLauncher/issues/513

**Comment by ghost** [comment]
I am not using an unstable version of GLFW, only AMDGPU and mesa. In fact Gentoo provides only one version of GLFW in the repos: [3.3.8](https://packages.gentoo.org/packages/media-libs/glfw)

I also do not use the "use native" override.

**Comment by Trayshar** [comment]
I have the same issue if I use the flatpak version of PrismLauncher 8. I tried the AppImage version and it worked fine. I have not made any changes to GLFW or any other library or setting.

Also, since the issue occurred after updating my graphics driver which I compile from source, I did some testing and bisected the issue to [this commit](https://gitlab.freedesktop.org/mesa/mesa/-/commit/e64ab3e4a94c4e8baee3bec3a5eb4d098b8e7193) in the mesa driver. I do not yet fully understand whats going on, but I think its a version mismatch between the mesa driver flatpak provides and the one I use natively. Likely not mesas fault. I'll look into it further the coming days.

**Comment by e1z0** [comment]
Add **MESA_GL_VERSION_OVERRIDE=4.3** to environment variables will fix it

**Comment by edricus** [comment]
> Add **MESA_GL_VERSION_OVERRIDE=4.3** to environment variables will fix it

doesn't fix the issue
Debian 12 
OpenGL version string: 4.6 (Compatibility Profile) Mesa 22.3.6

**Comment by HalJordan** [comment]
I had gotten this to go away. I believe I had to switch to the git version.

Lo and behold, the git version is now using flatpack AND IT BROKE AGAIN.

Why must we break what isn't already broken to go with the new shiny??? This is rediculous.

**Comment by HalJordan** [comment]
Yeah, definitely not limited to GPU manufacturer I have this issue now on amd, I had it before with nvidia, and someone else is seeing it now on nvidia in #1654. I'm able to run several other games/apps that rely on opengl INCLUDING using the POS put out by the FTB team. So it's not minecraft. It's not my drivers. It's something that the prism launcher is(not) doing.

This has happened before in the same way, then suddenly fixed itself but I was too busy with life to document when that happened.

I've tested with a handful of packs, so it's not down to any one pack or source(one imported from ftb, one from curseforge, etc).

**Comment by lukejameson** [comment]
As a few others have mentioned, installing a non flatpack version fixes it (At least for me).

Started getting the error after doing a system update (PosOs) which updated my GPU drivers from 470 to 550 (Yikes). 

Removed flat pack and installed via another method (Prebuilt MPR) and it all works now.

**Comment by LFF5644** [comment]
put `LIBGL_ALWAYS_SOFTWARE=1` in evn on launcher start arguments, disables you GPU and uses CPU software randering maybe slow but works

**Comment by ws909** [comment]
Also on Pop!_OS. 
Installed these updates from Cosmic Store, and that borked MC launched from Prism. The vanilla launcher launches the game just fine.
```
aug. 24 13:06:37 pull           org.freedesktop.Platform.GL.nvidia-550-67      1.4    user         flathub
aug. 24 13:06:39 deploy update  org.freedesktop.Platform.GL.nvidia-550-67      1.4    user         flathub
aug. 24 13:07:28 pull           org.freedesktop.Platform.GL.nvidia-555-58-02   1.4    user         flathub
aug. 24 13:07:30 deploy update  org.freedesktop.Platform.GL.nvidia-555-58-02   1.4    user         flathub
aug. 24 13:08:19 pull           org.freedesktop.Platform.GL32.nvidia-555-58-02 1.4    user         flathub
aug. 24 13:08:20 deploy update  org.freedesktop.Platform.GL32.nvidia-555-58-02 1.4    user         flathub
```

**Comment by thePalindrome** [comment]
Bumping because this is still an issue, and happened in the past few major versions (I'd updated prism and suddenly packs stop loading due to being unable to load the mesa driver .so files, regardless of java version specified as was suggested in a nixos bug report)

It happens to multiple minecraft versions, and was only an issue after updating (I *think* I was on 7.9 before). This is the portable download, so not flatpak sandboxing or anything

**Comment by thePalindrome** [comment]
Okay, after trying some things out, it seems that _Prism is corrupting instances_, creating a fresh portable install and reacquiring the pack (transferring saves and configs should be fine) is enough to fix it, although it would be a good idea to figure out what exactly is getting corrupted (nothing jumps out as modified settings or anything) and why.

I would suggest leaving this issue open until we can fully triage it, though

EDIT:
Okay, seems I was wrong, it launched right the first time, but now is refusing to work, even if I freshly unpack prism to a clean workspace, redownload the assets, libraries, and instance from scratch, and try then. It's literally just working the first time, then failing, which is incongruous with the suggestions of "Looking in the wrong spot for libraries" or "You have to use software rendering". What on earth is happening with this?

**Comment by thePalindrome** [comment]
Okay, found a fix.

go into your `lib` folder, and remove libstdc++.so.6 and libstdc++.so.6.0.28, and then it all just works

**Comment by HalJordan** [comment]
To clarify, you mean the lib directory within prism, correct?

**Comment by ct1735x** [comment]
> Okay, found a fix.
> 
> 
> 
> go into your `lib` folder, and remove libstdc++.so.6 and libstdc++.so.6.0.28, and then it all just works

Thx goat it worked amazingly

**Comment by thePalindrome** [comment]
>To clarify, you mean the lib directory within prism, correct?

Yes, the same place you'll find some other lib files like libQt<...>.so and whatnot.

Frustratingly, this isn't *entirely* a Prism issue, either, as libstdc++ is, well... "standard c-plus-plus library", and it's most likely *actually* dying because your system graphics driver libraries are an incompatible version to the bundled library. Deleting the file forces Prism to use your *system* library for it, which is naturally the right version.

Of course, the reason they have to bundle it is because your system library version might not match the libQt5 or libQt6 requirements, so it's just dependency headaches all around

**Comment by Saraistupid** [comment]
Having the same issue on Arch Linux under AMDGPU PRO driver, the game launches fine with Mesa drivers, tried the fixes mentioned above, the game still crashes with the same error, I also tried enabling `Use system installation of GLFW`.
```
    OpenGL Information
    GL_VENDOR:      ATI Technologies Inc.
    GL_RENDERER:    AMD Radeon RX 480 Graphics
    GL_VERSION:     4.6.0 Compatibility Profile Context .230614
```
```
[16:19:00] [Render thread/WARN]: Failed to create window: 
net.minecraft.class_1041$class_4716: GLFW error 65543: GLX: Failed to create context: GLXBadFBConfig
	at knot/net.minecraft.class_1041.method_4501(class_1041.java:221) ~[client-intermediary.jar:?]
	at knot/org.lwjgl.glfw.GLFWErrorCallbackI.callback(GLFWErrorCallbackI.java:43) ~[lwjgl-glfw-3.3.3.jar:?]
	at knot/org.lwjgl.system.JNI.invokePPPP(Native Method) ~[lwjgl-3.3.3.jar:?]
	at knot/org.lwjgl.glfw.GLFW.nglfwCreateWindow(GLFW.java:2058) ~[lwjgl-glfw-3.3.3.jar:?]
	at knot/org.lwjgl.glfw.GLFW.glfwCreateWindow(GLFW.java:2229) ~[lwjgl-glfw-3.3.3.jar:?]
	at knot/net.minecraft.class_1041.redirect$cdc000$sodium$wrapGlfwCreateWindow(class_1041.java:1108) ~[client-intermediary.jar:?]
	at knot/net.minecraft.class_1041.mixinextras$bridge$redirect$cdc000$sodium$wrapGlfwCreateWindow$61(class_1041.java) ~[client-intermediary.jar:?]
	at knot/net.minecraft.class_1041.wrapOperation$cak000$sodium$setAdditionalWindowHints(class_1041.java:593) ~[client-intermediary.jar:?]
	at knot/net.minecraft.class_1041.<init>(class_1041.java:103) ~[client-intermediary.jar:?]
	at knot/net.minecraft.class_3682.method_16038(class_3682.java:21) ~[client-intermediary.jar:?]
	at knot/net.minecraft.class_310.<init>(class_310.java:504) ~[client-intermediary.jar:?]
	at knot/net.minecraft.client.main.Main.main(Main.java:250) [client-intermediary.jar:?]
	at net.fabricmc.loader.impl.game.minecraft.MinecraftGameProvider.launch(MinecraftGameProvider.java:480) [fabric-loader-0.16.10.jar:?]
	at net.fabricmc.loader.impl.launch.knot.Knot.launch(Knot.java:74) [fabric-loader-0.16.10.jar:?]
	at net.fabricmc.loader.impl.launch.knot.KnotClient.main(KnotClient.java:23) [fabric-loader-0.16.10.jar:?]
	at org.prismlauncher.launcher.impl.StandardLauncher.launch(StandardLauncher.java:105) [NewLaunch.jar:?]
	at org.prismlauncher.EntryPoint.listen(EntryPoint.java:129) [NewLaunch.jar:?]
	at org.prismlauncher.EntryPoint.main(EntryPoint.java:70) [NewLaunch.jar:?]
```

**Comment by HalJordan** [comment]
@Saraistupid Do you have the box ticked for use discreet gpu? Whichever you have set, flip it the other way and try.

**Comment by Saraistupid** [comment]
Tried both on and off, game still crashes with the same log error.

**Comment by reclaimerx10** [comment]
Just got this same error on my linux mint machine. whats the fix? it is a Samsung SF410 laptop and just rebooted it. never gotten it to work, just want to get it running

**Comment by strixdio** [comment]
I'm getting this on my raspberry pi 4, fresh install of raspian and prismlauncher. So far, nothing has gotten it to run.

**Comment by oxygencamera** [comment]
I have the same issue on Fedora Linux 42 and nothing that's been mentioned above fixes it.

**Comment by oxygencamera** [comment]
> I have the same issue on Fedora Linux 42 and nothing that's been mentioned above fixes it.

Nevermind, it fixed itself with the latest Mesa update.

**Comment by SwayvPKG** [comment]
If none of the tricks work for you, try this:
Create a script or configure your launcher’s wrapper to delete the file `iris-excluded.json` before launching Minecraft.

`rm -f config/iris-excluded.json`

**Comment by erstert3st** [comment]
@tabbeddev thx

**Comment by Ari-43** [comment]
Switching from using the `[X] Use discrete GPU` option to manually setting `DRI_PRIME=1` as an environment variable worked around this for me. This will only work on AMD dGPUe/eGPUs.

**Comment by FifthTundraG** [comment]
Running Arch I find myself running into this issue quite often (every 1-2 months, sometimes even more frequent), but the solution has consistently been the same for me, which is to update my graphics drivers. Just saying that isn't contributing much though, as it's been mentioned before, but what I found very confusing when I first encountered this issue is that a generic `pacman -Syu` (or `apt update && apt update` for my APT friends) would never cut it. The distinction that confused me was this:

Flatpak packages NVIDIA drivers separately. As such, updating your _system_ packages would not update the _flatpak_ packages. At the time I pretty consistently used Plasma Discover for updating my flatpaks, and they don't highlight updates for "system" flatpaks as much as they do application ones, so I never noticed it. Only when I started using the command line for flatpak updates did I realize that NVIDIA drivers were packaged there, and even later did I connect the dots of this error message to that issue.

tl;dr, if you're like me and get this problem, running these commands + a reboot should (hopefully) do it for ya:

1. `sudo pacman -Syu` (or whatever the "update" function on your package manager is; this is _maybe_ not necessary but it's good to cover all bases)
2. `flatpak update` (or `sudo flatpak update` depending on install method)

**Comment by ell1e** [comment]
What's the cause for this for non-NVidia systems, though? I'm seeing this on a Raspberry Pi 5, for example.

**Comment by YeetElite** [comment]
> However doing a complete reinstall and deleting the configs made in `~/.var` made it work again. (Flatpak)

This happened to me while my game was running, just deleting the config folder in `~/.var/app/...` worked.
